### PR TITLE
Jakarta MVC 1.1

### PR DESCRIPTION
**Pull request by ivargrimstad**
Signed-off-by: Ivar Grimstad <ivar.grimstad@eclipse-foundation.org>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x]  For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
N/A
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/mvc/jakarta.mvc-api/1.1.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/mvc/staged/1.1/jakarta-mvc-tck-1.1.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/mvc-api/issues/45
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by kwsutter** [comment]
I volunteered to mentor the MVC Spec release...  Not everything in the Jakarta EE checklist is necessarily applicable with MVC, but it's still a good place to start....

# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by ivargrimstad** [comment]
@kwsutter: This PR should be ready now.

**Review by kwsutter — CHANGES_REQUESTED** [review]
The README and the TCK instructions are not in sync.  One says to run "mvn verify" and the other says "mvn install test".  I am also confused by the CI.  Your CR references Krazo.  But, the instructions seems to imply that Glassfish is the CI.  Maybe it's Glassfish plus your Krazo plugin?  Can you clarify these items?

**Comment by ivargrimstad** [comment]
> The README and the TCK instructions are not in sync. One says to run "mvn verify" and the other says "mvn install test". I am also confused by the CI. Your CR references Krazo. But, the instructions seems to imply that Glassfish is the CI. Maybe it's Glassfish plus your Krazo plugin? Can you clarify these items?

I guess the README should say `mvn verify` as well. I will also add a comment that this is a "short version" and the user guide should be referred to as the complete setup.

So to the GlassFish question. Krazo is the implementation, but cannot run standalone. It has to run in a Jakarta EE Application server. GlassFish is in this case an example. When I ran the TCK, I used Wildfly (which the TCK summary shows). 
The TCK use Arqulliían for the tests, and the way it is set up requires a Jakarta EE application server to be running as well. I will add to the README that the "short version is just an example".

File: mvc/1.1/_index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* [Jakarta MVC 1.1 Specification Document](./jakarta-mvc-spec-1.1.pdf) (PDF)
+* [Jakarta MVC 1.1 Specification Document](./jakarta-mvc-spec-1.1.html) (HTML)
+* [Jakarta MVC 1.1 Javadoc](./apidocs)
+* [Jakarta MVC 1.1 TCK](http://downloads.eclipse.org/jakartaee/mvc/1.1/jakarta-mvc-tck-1.1.0.zip)
+* Maven coordinates
+  * [jakarta.mvc:jakarta.mvc-api:jar:1.1.0](https://search.maven.org/artifact/jakarta.mvc/jakarta.mvc-api/1.1.0/jar)
+
+# Compatible Implementations
+
+* [Eclipse Krazo 1.1.0](https://projects.eclipse.org/projects/ee4j.krazo)
```
[the remark this answers is not in the document]
Comment (ivargrimstad) [reply mvc/1.1/_index.md:18]: The way you "download" Krazo is by adding dependencies in the applications's pom.xml file. Which dependencies depends on what platform you target. Currently Jersey, RestEasy and CXF are supported. I guess the easiest way is to link to a "usage" page on the Krazo GitHub pages that explains how to do it?

[the remark this answers is not in the document]
Comment (ivargrimstad) [reply mvc/1.1/_index.md]: Fixed!

**Comment by ivargrimstad** [comment]
> > The README and the TCK instructions are not in sync. One says to run "mvn verify" and the other says "mvn install test". I am also confused by the CI. Your CR references Krazo. But, the instructions seems to imply that Glassfish is the CI. Maybe it's Glassfish plus your Krazo plugin? Can you clarify these items?
> 
> I guess the README should say `mvn verify` as well. I will also add a comment that this is a "short version" and the user guide should be referred to as the complete setup.
> 
> So to the GlassFish question. Krazo is the implementation, but cannot run standalone. It has to run in a Jakarta EE Application server. GlassFish is in this case an example. When I ran the TCK, I used Wildfly (which the TCK summary shows).
> The TCK use Arqulliían for the tests, and the way it is set up requires a Jakarta EE application server to be running as well. I will add to the README that the "short version is just an example".

Clarified in new staged version of TCK

**Comment by kwsutter** [comment]
> Clarified in new staged version of TCK.

Thanks, @ivargrimstad!  TCK looks better, but now the SHA values don't match up...

**Comment by ivargrimstad** [comment]
Fixed!

**Review by kwsutter — APPROVED** [review]
Looks good, @ivargrimstad!  Thanks!

**Comment by kwsutter** [comment]
* On ballot completion, the specification committee mentor:
 - [x] adds this final checklist to the main PR.
 - [x] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>


 ----
 - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
 - [ ] approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] merge any final release branch as appropriate for the branch management for the project.

**Comment by kwsutter** [comment]
@ivargrimstad The TCK Results existed at one time (if I can trust my earlier comments).  But, now I can't seem to find them.  The CR has this link, but it doesn't work:  https://eclipse-ee4j.github.io/krazo/certifications/jakarta-mvc/1.1/TCK-results.html

Can you get this corrected?  I will hold off on merging this PR until we figure out happened to the TCK Results.  Thanks!

**Comment by ivargrimstad** [comment]
> @ivargrimstad The TCK Results existed at one time (if I can trust my earlier comments). But, now I can't seem to find them. The CR has this link, but it doesn't work: https://eclipse-ee4j.github.io/krazo/certifications/jakarta-mvc/1.1/TCK-results.html
> 
> Can you get this corrected? I will hold off on merging this PR until we figure out happened to the TCK Results. Thanks!

It's a capital R in Results that was missing. It is fixed now: https://eclipse-ee4j.github.io/krazo/certifications/jakarta-mvc/1.1/TCK-Results.html

**Comment by keilw** [comment]
Is 1.1 likely to be deployed to Maven repositories before the end of the week or could it take a bit more?

**Comment by kwsutter** [comment]
> Is 1.1 likely to be deployed to Maven repositories before the end of the week or could it take a bit more?

You will need to follow the MVC issue:  https://github.com/eclipse-ee4j/mvc-api/issues/51

**Pull request merged**